oci/layout/oci_dest.go

```diff
@@ -231,3 +238,30 @@ func (d *ociImageDestination) Commit() error {
 	}
 	return ioutil.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
+
+func ensureDirectoryExists(path string) error {
+	if _, err := os.Stat(path); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(path, 0755); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ensureParentDirectoryExists ensures the parent of the supplied path exists.
+func ensureParentDirectoryExists(path string) error {
+	return ensureDirectoryExists(filepath.Dir(path))
+}
+
+// Exists checks whether an index file exists at the specified path.
+// The implementation is opinionated, since in case of unexpected errors false is returned
+func indexExists(path string) bool {
+	_, err := os.Stat(filepath.Join(path, "index.json"))
```

Comment:
This should call the existing `ociReference.indexPath`, at least to make it trivial to find all uses of that path.

Reply:
Not sure what you mean here

Reply:
```go
func (ref ociReference) indexExists() bool {
    _, err := os.Stat(ref.indexPath())
   …
```

Reply:
ahh :-)